### ISSUE TITLE
missing config causes panic

### ISSUE THREAD

**Issue by tgross** [joyent/containerpilot#382]
While trying a local test I gave `-reload` a missing file and got a panic. Obviously we can't do anything anyways here but we should return a friendly error message.

```
./build/containerpilot -config nonexisting.json -reload
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13a24af]

goroutine 1 [running]:
github.com/joyent/containerpilot/core.LoadApp(0x1, 0x8, 0xc420053f68)
        /Users/tim.gross/src/joyent/containerpilot/src/github.com/joyent/containerpilot/core/app.go:117 +0x6cf
main.main()
        /Users/tim.gross/src/joyent/containerpilot/src/github.com/joyent/containerpilot/main.go:16 +0x33
```

**Comment by tgross** [comment]
The problem is pretty tiny. [Here]( https://github.com/joyent/containerpilot/blob/9b4ff6b40d630b5d27d39cf8333aa54a3c9cec36/core/app.go#L114) we ignore the error if no config flag is passed in, but that means `cmd` was set to `nil` and we get a NPE when we try to use it.

But once I fixed that, I realized that we're loading the configuration whether or not a subcommand is in play, which means we're loading the configuration twice in cases where we're not. So I'll fix that in this PR.
